src/core/Akka.Remote/Serialization/LruBoundedCache.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Akka.Remote.Serialization
+{
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <remarks>
+    /// Fast hash based on the 128 bit Xorshift128+ PRNG. Mixes in character bits into the random generator state.
+    /// </remarks>
+    public static class FastHash
+    {
+        public static int OfString(string s)
+        {
+            var chars = s.ToCharArray();
+            var s0 = 391408L;
```

Comment:
The magic numbers are just seed values, since the hash won't work properly if it's seeded at 0 initially: https://en.wikipedia.org/wiki/Xorshift 

In terms of why these exact seed numbers were chosen, no idea. Nothing special about them.